### PR TITLE
Add container management procedure

### PR DESCRIPTION
**Pull request by yokaze**
Signed-off-by: Daichi Sakaue <daichi-sakaue@cybozu.co.jp>

### CODE REVIEW

File: maintenance.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    - `squid`  
+      Keep track of [the latest apt package version](https://launchpad.net/ubuntu/+source/squid3).
+      This container is deprecated and will be removed sometime.
+    - `vault`
+
+### Notes
+- Most of the softwares provide the release note on GitHub, but some do not.
+  Their release notes are pointed in the above list.
+- If only the Ubuntu base image is updated while the software version is decided to remain unchanged, increment the last number of the image tag to create the newer image like `.2` and `.3`. 
+- If build fails, consult the upstream `Dockerfile` to keep up with the content.
```
Comment (zoetrope) [inline maintenance.md:55]: We have to check it all the time, not just when build fails.

File: maintenance.md [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+
+1. Check the [release page](https://github.com/envoyproxy/envoy/releases).
+2. Update the `version` argument on the `build-envoy` job in the CircleCI `main` workflow.
+3. Update `BAZEL_VERSION` in `build-envoy` job. The required version is written in the following file.
+   - https://github.com/envoyproxy/envoy/blob/vX.Y.Z/.bazelversion
+4. Update image tag in `README.md`.
+5. Update `BRANCH` and `TAG` files.
+
+## etcd
+
```
Comment (tapih) [inline maintenance.md:214]: I understand that both the Kubernetes badge and the regular update badge are not needed here, but it's better to write etcd is updated in a separated task.

**Review by tapih — CHANGES_REQUESTED** [review]

File: maintenance.md [hunk 3]
```diff
@@ -0,0 +1,518 @@
+How to maintain neco-containers
+===============================
+
+This document describes the procedure for updating each container image.
+
+Besides on-demand container updates, we have two regular renewal operations: `Kubernetes Update` and `Regular Update`.
+The target container of these operations have the following badges, so check before the operations.
+
```
[the remark this answers is not in the document]
Comment (masa213f) [reply maintenance.md:8]: I wrote in a different document.

**Review by tapih — APPROVED** [review]
LGTM

**Pull request merged**